services/snapshotter/service.go

```diff
@@ -274,12 +359,46 @@ func (s *Service) writeRetentionPolicyInfo(conn net.Conn, database, retentionPol
 }
 
 // readRequest unmarshals a request object from the conn.
-func (s *Service) readRequest(conn net.Conn) (Request, error) {
+func (s *Service) readRequest(conn net.Conn) (Request, []byte, error) {
 	var r Request
-	if err := json.NewDecoder(conn).Decode(&r); err != nil {
-		return r, err
+	d := json.NewDecoder(conn)
+
+	if err := d.Decode(&r); err != nil {
+		return r, []byte{}, err
 	}
-	return r, nil
+
+	bits := make([]byte, r.UploadSize+1)
+
+	if r.UploadSize > 0 {
+
+		remainder := d.Buffered()
+
+		//// no time to investigate.  It seems the JSON encoder puts one extra byte on the stream,
```

Comment:
What is going on here?

Reply:
OK, I just read through... and it looks like at one point the protocol is `[byte].[byte].[JSON-object].[binary-data]`... That seems far too fragile to be useful or maintainable.